Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -379,10 +382,30 @@ void WebFrameProxy::didCreateSubframe(WebCore::FrameIdentifier frameID)
     m_childFrames.add(WTFMove(child));
 }
 
-void WebFrameProxy::swapToProcess(Ref<WebProcessProxy>&& process, const WebCore::ResourceRequest& request)
+void WebFrameProxy::prepareForProvisionalNavigationInProcess(WebProcessProxy& process, CompletionHandler<void()>&& completionHandler)
 {
     ASSERT(!isMainFrame());
-    m_provisionalFrame = makeUnique<ProvisionalFrameProxy>(*this, WTFMove(process), request);
+
+    if (m_provisionalFrame && m_provisionalFrame->process().processID() == process.processID())
+        return completionHandler();
+
+    if (!m_provisionalFrame) {
+        m_provisionalFrame = makeUnique<ProvisionalFrameProxy>(*this, Ref { process });
+        // FIXME: This gives too much cookie access. This should be removed when a RemoteFrame is given a topOrigin member.
+        auto giveAllCookieAccess = LoadedWebArchive::Yes;
+        WebCore::RegistrableDomain domain { };
+        page()->websiteDataStore().networkProcess().sendWithAsyncReply(Messages::NetworkProcess::AddAllowedFirstPartyForCookies(process.coreProcessIdentifier(), domain, giveAllCookieAccess), WTFMove(completionHandler));
+    }
+
+    if (m_process->processID() != process.processID()) {
+        LocalFrameCreationParameters localFrameCreationParameters {
+            std::optional { m_provisionalFrame->layerHostingContextIdentifier() }
```

Comment:
nit: Why explicitly wrap this in std::optional, won't it do the right thing without it?

Reply:
An earlier version used a ternary operator.  You're right.  It's not necessary.